source/server/server.cc

```diff
@@ -493,8 +491,8 @@ void InstanceImpl::terminate() {
     flushStats();
   }
 
-  if (config_ != nullptr && config_->clusterManager() != nullptr) {
-    config_->clusterManager()->shutdown();
+  if (config_.clusterManager() != nullptr) {
```

Comment:
I'm not sure how this can be a safe change if we checked for nullptr before. Are you sure that cluster manager exists at this point? This is related to my comment above, but is this change from interface to impl needed in this change?

Reply:
config_ is now a direct object and not a unique_ptr so this is safe and IMO marginally simpler.

However you are right that this change is not critical to this PR and I am happy to back it out if you prefer.

Reply:
actually I went ahead and backed it out. I'll issue it as a separate PR.

Reply:
OK I remember why I messed with this now. clang-tidy doesn't like it the way the code is in the repo now. Because it grandfathers in old non-compliant code it was OK until I made edits near it, and my edit makes it clang-tidy clean.

Also it's super-annoying to iterate with clang-tidy as it takes 40 minutes to run locally for me under docker, and it's non-deterministic in CI.

Here's the conundrum:
 - clang-tidy doesn't like naked calls to new, with Config* variables.
 - the semantics of the code as is require two pointers to the same memory: one owned
    by Server that's got the interface type, and one local that has the Impl type. The Impl
    type provides an initialize() method and the interface does not.
 - The implementation of the configure method requires that server->config_ be populated
Without a more significant refactoring I don't think it's possible to make this code work and also make clang-tidy happy.

Note that not much of this has anything to do with the Http::Context propagation or precalculation of StatsCode symbols, but I must've woken up clang-tidy by editing code nearby.

So I think there are a few choices here:
 - merge even though there are clang-tidy failures
 - do a more significant refactoring of the initialization() dance here
 - move forward with #5184 despite @venilnoronha's concern that this adds another impl dependency to server.h

My vote is to move forward with #5814 -- I think it's the cleanest, most honest approach that doesn't require a significant refactor.